### PR TITLE
ARM64 support

### PR DESCRIPTION
**Pull request by tybritten**

### CODE REVIEW

**Comment by tybritten** [comment]
1. updates all binary and docker builds to build both AMD64/ARM64
2. runs all tests in the `circleci` workflow on both platforms
3. updates tests to use multiarch images if they're using other external images (MySQL, etc)

File: etc/testing/circle/install.sh [hunk 1]
```diff
@@ -52,32 +59,24 @@ fi
 # To get the latest etcd version:
 # curl -Ls https://api.github.com/repos/etcd-io/etcd/releases | jq -r .[].tag_name
 if [ ! -f cached-deps/etcdctl ] ; then
-    ETCD_VERSION=v3.3.12
-    curl -L https://storage.googleapis.com/etcd/${ETCD_VERSION}/etcd-${ETCD_VERSION}-linux-amd64.tar.gz \
+    ETCD_VERSION=v3.5.1
```
Comment (nitinjainsj) [inline etc/testing/circle/install.sh:62]: 3.5.2?

File: Makefile [hunk 2]
```diff
@@ -115,16 +148,21 @@ docker-gpu: docker-build-gpu docker-push-gpu
...
 
 docker-push: docker-tag
-	$(SKIP) docker push pachyderm/pachd:$(VERSION)
-	$(SKIP) docker push pachyderm/worker:$(VERSION)
-	$(SKIP) docker push pachyderm/pachctl:$(VERSION)
-	$(SKIP) docker push pachyderm/pachtf:$(VERSION)
+	$(SKIP) docker push pachyderm/pachd:$(VERSION)-$(TARGET_ARCH)
+	$(SKIP) docker push pachyderm/worker:$(VERSION)-$(TARGET_ARCH)
+	$(SKIP) docker push pachyderm/pachctl:$(VERSION)-$(TARGET_ARCH)
+	#$(SKIP) manifest create pachyderm/pachd:$(VERSION) docker.io/pachyderm/pachd:$(VERSION)-amd64 docker.io/pachyderm/pachd:$(VERSION)-arm64
```
Comment (nitinjainsj) [inline Makefile:160]: pachtf?

Remove all the manifest commands

File: Makefile [hunk 3]
```diff
@@ -115,16 +148,21 @@ docker-gpu: docker-build-gpu docker-push-gpu
...
 
 docker-tag:
-	docker tag pachyderm/pachd pachyderm/pachd:$(VERSION)
-	docker tag pachyderm/worker pachyderm/worker:$(VERSION)
-	docker tag pachyderm/pachctl pachyderm/pachctl:$(VERSION)
-	docker tag pachyderm/pachtf pachyderm/pachtf:$(VERSION)
+	docker tag pachyderm/pachd:latest pachyderm/pachd:$(VERSION)-$(TARGET_ARCH)
+	docker tag pachyderm/worker:latest pachyderm/worker:$(VERSION)-$(TARGET_ARCH)
+	docker tag pachyderm/pachctl:latest pachyderm/pachctl:$(VERSION)-$(TARGET_ARCH)
+
```
Comment (nitinjainsj) [inline Makefile:154]: pachtf?

File: Makefile (hunk 2 above)
Comment (tybritten) [reply Makefile:160]: i'll add the pachtf lines..why are we removing the manifest lines?

Comment (nitinjainsj) [reply Makefile:161]: These seem like permanently skipped. If so, remove the dead code.

**Comment by smalyala** [comment]
We should probably add a note on how to create future versions of multiarch images like pgbouncer and etcd (i.e. explaining that a release in the GH repo will automatically build and push multiarch image to DH). Is that best added in the core pach repo or copy and pasted in the various repos themselves?

**Comment by tybritten** [comment]
I'd think that would be in the readmes of those individual repos

**Review by nitinjainsj — COMMENTED** [review]
Couple of more comments.

File: Makefile [hunk 4]
```diff
@@ -115,16 +148,24 @@ docker-gpu: docker-build-gpu docker-push-gpu
...
+	$(SKIP) manifest create pachyderm/worker:$(VERSION) docker.io/pachyderm/worker:$(VERSION)-amd64 docker.io/pachyderm/worker:$(VERSION)-arm64
+	$(SKIP) docker manifest create pachyderm/pachctl:$(VERSION) docker.io/pachyderm/pachctl:$(VERSION)-amd64 docker.io/pachyderm/pachctl:$(VERSION)-arm64
+	$(SKIP) docker manifest create pachyderm/pachtf:$(VERSION) docker.io/pachyderm/pachtf:$(VERSION)-amd64 docker.io/pachyderm/pachtf:$(VERSION)-arm64
+	$(SKIP) docker manifest push pachyderm/worker:$(VERSION)
+	$(SKIP) docker manifest push pachyderm/worker:$(VERSION)
+	$(SKIP) docker manifest push pachyderm/pachctl:$(VERSION)
+	$(SKIP) docker manifest push pachyderm/pachtf:$(VERSION)
 
 docker-push-release: docker-push
 	$(SKIP) docker push pachyderm/etcd:v3.5.1
```
[the remark this answers is not in the document]
Comment (tybritten) [reply Makefile:171]: no, they're pushed from their own repos

[the remark this answers is not in the document]
Comment (tybritten) [reply Makefile:171]: same with etcd..its own repo

**Pull request closed without merging**